From 4c4dbde3be7325f5798bd22b97cef39718a1d550 Mon Sep 17 00:00:00 2001
From: Asutosh Mohapatra <asutmoha@qti.qualcomm.com>
Date: Wed, 30 Apr 2025 02:42:41 -0700
Subject: [PATCH 25/40] qcacld-3.0: Add Validation for WMA Handle and PSOC in
 Wake Event

Currently, the WOW wakeup event handler lacks validation for the
WMA handle and the PSOC pointer within the WMA handle. This omission
can lead to null pointer dereferences in the host.

To address this issue, null pointer checks for both the WMA handle
and the PSOC pointer have been added.

CRs-Fixed: 4107000
Change-Id: Iaf22d5adc14b65b778b0e1d78108eded0cccb8c9
---
 core/wma/src/wma_features.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/core/wma/src/wma_features.c b/core/wma/src/wma_features.c
index 973beab8e3..0f062defee 100644
--- a/core/wma/src/wma_features.c
+++ b/core/wma/src/wma_features.c
@@ -3812,6 +3812,9 @@ int wma_wow_wakeup_host_event(void *handle, uint8_t *event, uint32_t len)
 	WMI_WOW_WAKEUP_HOST_EVENTID_param_tlvs *event_param;
 	WOW_EVENT_INFO_fixed_param *wake_info;
 
+	if (!wma || !wma->psoc)
+		return -EINVAL;
+
 	event_param = (WMI_WOW_WAKEUP_HOST_EVENTID_param_tlvs *)event;
 	if (!event_param) {
 		wma_err("Wake event data is null");
-- 
2.34.1

